include/grpc/grpc_security.h

```diff
@@ -131,6 +131,34 @@ grpc_credentials *grpc_google_iam_credentials_create(
     const char *authorization_token, const char *authority_selector,
     void *reserved);
 
+/* Function to be called by the metadata credentials plugin implementation when
+   the metadata is ready. */
+void grpc_credentials_plugin_metadata_notify(
```

Comment:
The problem with `grpc_metadata_credentials_plugin_notify` is that this implies that this is a method of the `grpc_metadata_credentials_plugin` (you would expect a pointer to this type as the first param).
Howe about `grpc_metadata_credentials_notify_from_plugin`